miio/push_server/serverprotocol.py

```diff
@@ -52,57 +52,101 @@ def send_ping_ACK(self, host, port):
         self.transport.sendto(m, (host, port))
         _LOGGER.debug("%s:%s<=ACK(server_id=%s)", host, port, self.server.server_id)
 
-    def send_msg_OK(self, host, port, msg_id, token):
-        # This result means OK, but some methods return ['ok'] instead of 0
-        # might be necessary to use different results for different methods
-        result = {"result": 0, "id": msg_id}
+    def _create_message(self, data, token, device_id):
+        """Create a message to be sent to the client."""
         header = {
             "length": 0,
             "unknown": 0,
-            "device_id": self.server.server_id,
+            "device_id": device_id,
             "ts": datetime.datetime.now(),
         }
         msg = {
-            "data": {"value": result},
+            "data": {"value": data},
             "header": {"value": header},
             "checksum": 0,
         }
         response = Message.build(msg, token=token)
-        self.transport.sendto(response, (host, port))
+
+        return response
+
+    def send_response(self, host, port, msg_id, token, payload=None):
+        if payload is None:
+            payload = {}
+
+        result = {**payload, "id": msg_id}
```

Comment:
I would call this data instead of result

Reply:
Done :+1: